### PR TITLE
[#1088] PDF Page Counts For Some Records Is Off

### LINKED ISSUE #1088: [Export] PDF page count for some records is off
This is an error both in DH and Prism, so possibly a bug with the file. PDF file page count in both DH and Prism is "0 pages", but this PDF has 44 pages.

https://vtfsmghslrepo02.fsm.northwestern.edu/records/9zh4q-y4k87
https://digitalhub.northwestern.edu/files/9cd9e3df-458c-4a2c-9e42-1dcdc95e7adf

Screenshot of DH (it's in admin data):
image

### PR DESCRIPTION
**Pull request by Meowcenary**
Specify page count for records with erroneous pdf page counts. closes #1088

### CODE REVIEW

File: app/models/invenio_rdm_record_converter.rb [hunk 1]
```diff
@@ -611,4 +611,14 @@ def publisher
   def pediatric_neurology_brief?
     @generic_file.depositor == INSTITUTIONAL_PNB_DEPOSITOR || @generic_file.doi.shift.include?(PNB_DOI_PREFIX)
   end
+
+  def sizes
+    if @generic_file.id == "3105875f-61e1-412c-9b1c-c8a33b37ff35"
```
Comment (Meowcenary) [inline app/models/invenio_rdm_record_converter.rb:616]: This seems like the fastest and simplest solution. If we find more situations like this we can just add more elif statements.

File: app/models/invenio_rdm_record_converter.rb [hunk 2]
```diff
@@ -611,4 +611,14 @@ def publisher
...
+      ["116 pages"]
+    elsif  @generic_file.id == "9cd9e3df-458c-4a2c-9e42-1dcdc95e7adf"
+      ["44 pages"]
+    elsif @generic_file.page_count
+      ["#{@generic_file.page_count} pages"]
```
Comment (fenekku) [inline app/models/invenio_rdm_record_converter.rb:621]: This is another issue with pages that we should fix: `@generic_file.page_count` is an array so the final string looks like:

`'["0"] pages'`

when it should be:

`'0 pages'`

(e.g., https://vtfsmghslrepo02.fsm.northwestern.edu/records/653g4-8m023)
Although these 0 pages shouldn't really be the norm actually - was `@generic_file.page_count.blank?` discarded for a specific reason?

Popping the page count from `page_count` should probably be enough.

Comment (Meowcenary) [reply app/models/invenio_rdm_record_converter.rb:621]: Oh, I think that was actually on the previous version. I was wondering why that was there.

Comment (Meowcenary) [reply app/models/invenio_rdm_record_converter.rb]: Okay, updated!

File: app/models/invenio_rdm_record_converter.rb [hunk 3]
```diff
@@ -611,4 +611,14 @@ def publisher
   def pediatric_neurology_brief?
     @generic_file.depositor == INSTITUTIONAL_PNB_DEPOSITOR || @generic_file.doi.shift.include?(PNB_DOI_PREFIX)
   end
+
+  def sizes
+    if @generic_file.id == "3105875f-61e1-412c-9b1c-c8a33b37ff35"
+      ["116 pages"]
+    elsif  @generic_file.id == "9cd9e3df-458c-4a2c-9e42-1dcdc95e7adf"
+      ["44 pages"]
+    elsif @generic_file.page_count
```
Comment (sharpattack) [inline app/models/invenio_rdm_record_converter.rb:620]: You need to check if this is blank, right?

Comment (Meowcenary) [reply app/models/invenio_rdm_record_converter.rb:620]: Nope, we want to check if there is anything to format. If it's blank we will just skip it.

Comment (sharpattack) [reply app/models/invenio_rdm_record_converter.rb:620]: Whoops, I meant not blank. Empty arrays are true, right?

Comment (Meowcenary) [reply app/models/invenio_rdm_record_converter.rb:620]: oh right! that was the reason it was using that logic

**Review by fenekku — APPROVED** [review]
Updates look good to me. Thanks!

**Pull request merged**